apps/webapp/src/state/send.ts

```diff
@@ -56,6 +60,23 @@ export const createSendSlice = (): SliceCreator<SendSlice> => (set, get) => {
         state.send.memo = txt;
       });
     },
+    refreshFee: async () => {
+      const { amount, recipient, selection, memo } = get().send;
+
+      if (!amount || !recipient || !selection) {
+        set(state => {
+          state.send.fee = undefined;
+        });
+        return;
+      }
+
+      const plan = await getPlan({ amount, recipient, selection, memo });
+      if (!plan?.fee?.amount) return;
+
+      set(state => {
+        state.send.fee = plan.fee;
+      });
+    },
```

Comment:
planning should be quick, but it does spin up a wasm runtime and make idb queries. input events can trigger very rapidly, so the request should be debounced to avoid hammering the endpoint. 

give it a largeish subsecond delay. if it's easy, apply CallOptions with an abort signal - we don't respect the signal in the service yet, but that should happen eventually.

https://github.com/connectrpc/connect-es/blob/main/packages/connect/src/call-options.ts#L38

Reply:
you could debounce either here or in the ui component

Reply:
@turbocrime good call — I actually ended up debouncing it [in the `useRefreshFee` hook](https://github.com/penumbra-zone/web/pull/362/files#diff-7676de151de670030b837855d561ce6f0a44fb45c3c31c086c033503c77d59ef).

Reply:
Aaand I forgot to use the abort signal — will fix.

Reply:
Per discussion in today's web weekly meeting, we're going to skip using the abort signal for now, especially since planning is so fast.